tensorflow/python/keras/utils/vis_utils.py

```diff
@@ -227,7 +273,9 @@ def format_shape(shape):
       dot.add_node(node)
 
   # Connect nodes with edges.
-  for layer in layers:
+  for i, layer in enumerate(layers):
+    if i <= layer_range[0] or i >= layer_range[1]:
```

Comment:
For "<=" here, is that intended? We might miss the first layer in the range.

Reply:
@qlzh727 , for nodes we have `<` which is another loop above this loop. This loop is for tracking edges. if we put `<` here as well, then it will look for the edge (before the first node in range) which is not part of subgraph which we want to plot and thus the method will fail. To avoid that I am starting the edge tracking from 2nd node in our range so that it will consider all edges in subgraph that we want. This method works fine. You can see it working as expected in this [gist](https://colab.research.google.com/drive/1ynkoCxW9RPh8s0HgUAo_N0pNDrUXOqKk?usp=sharing)